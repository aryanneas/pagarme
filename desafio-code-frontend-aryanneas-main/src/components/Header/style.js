import styled from 'styled-components'
import { sizes } from '../../utils/responsive'

export const HeaderWrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 2%;
  background-color: #f2f2f3;

  @media (max-width: ${sizes.md}) {
    padding: 5%;
  }
`

export const Title = styled.p`
  color: #1d1647;
  font-size: 16px;
  margin: auto;
`
