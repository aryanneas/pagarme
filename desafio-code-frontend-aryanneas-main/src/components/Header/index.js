import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import BackIcon from '../../assets/img/backIcon.png'

import { HeaderWrapper, Title } from './style'

const Header = ({ title }) => {
  return (
    <nav>
      <HeaderWrapper>
        <Link to="/">
          <img src={BackIcon} alt="voltar" />
        </Link>

        <Title>{title}</Title>
      </HeaderWrapper>
    </nav>
  )
}
Header.propTypes = {
  title: PropTypes.string,
}

export default Header
