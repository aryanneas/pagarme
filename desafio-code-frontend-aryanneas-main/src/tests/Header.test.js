import React from 'react'
import { render } from '@testing-library/react'

import { BrowserRouter, Switch, Route } from 'react-router-dom'

import Header from '../components/Header'

const title = "Hi! I'm a header title"

describe('Header component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Switch>
          <Route>
            <Header title={title} />
          </Route>
        </Switch>
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
