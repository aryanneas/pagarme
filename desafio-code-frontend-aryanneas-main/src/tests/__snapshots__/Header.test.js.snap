// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="sc-bdfBwQ lhdpVo"
    >
      <a
        href="/"
      >
        <img
          alt="voltar"
          src="backIcon.png"
        />
      </a>
      <p
        class="sc-gsTCUz fcjGHL"
      >
        Hi! I'm a header title
      </p>
    </div>
  </nav>
</DocumentFragment>
`;
